src/pages/post-outwards/index.js

```diff
@@ -0,0 +1,251 @@
+  const { platformLabels } = useContext(ControlContext)
+
+  const {
+    labels: _labels,
+    access,
+    refetch
+  } = useResourceQuery({
+    endpointId: RemittanceOutwardsRepository.Postoutwards.qry,
+    datasetId: ResourceIds.PostOutwards
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      countryId: '',
+      currencyId: '',
+      corId: '',
+      dispersalType: ''
+    },
+    access,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      countryId: yup.string().required()
+    }),
+    onSubmit: async values => {
+      const checkedUserIds = data.filter(row => row.checked).map(row => row.recordId)
+
+      if (checkedUserIds.length > 0) {
+        try {
+          await postRequest({
+            extension: RemittanceOutwardsRepository.Postoutwards.post2,
+            record: JSON.stringify({ ids: checkedUserIds })
+          })
+        } catch (error) {}
+      }
+      fetchRemittanceData(
+        formik.values.countryId,
+        formik.values.currencyId,
+        formik.values.corId,
+        formik.values.dispersalType
+      )
+      toast.success(platformLabels.Posted)
+    }
+  })
+
+  const rowColumns = [
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'date',
+      headerName: _labels.date,
+      flex: 1,
+      type: 'date'
+    },
+
+    {
+      field: 'currencyRef',
+      headerName: _labels.currency,
+      flex: 1
+    },
+    {
+      field: 'corName',
+      headerName: _labels.cor,
+      flex: 1
+    },
+    {
+      field: 'clientName',
+      headerName: _labels.client,
+      flex: 1
+    },
+    {
+      field: 'amount',
+      headerName: _labels.amount,
+      flex: 1,
+      type: 'number'
+    }
+  ]
+
+  const fetchRemittanceData = () => {
+    if (!formik.values.countryId) return
+
+    getRequest({
+      extension: RemittanceOutwardsRepository.Postoutwards.qry,
+      parameters: `_countryId=${formik.values.countryId}&_currencyId=${formik.values.currencyId || 0}&_corId=${
+        formik.values.corId || 0
+      }&_dispersalType=${formik.values.dispersalType || 0}`
+    })
+      .then(response => {
+        setData(response.list || [])
+      })
+      .catch(error => {})
+  }
+
+  useEffect(() => {
+    const { countryId, currencyId, corId, dispersalType } = formik.values
+    if (!!countryId) {
+      fetchRemittanceData(countryId, currencyId, corId, dispersalType)
+    }
+  }, [formik.values.countryId, formik.values.currencyId, formik.values.corId, formik.values.dispersalType])
+
+  console.log(formik.values)
+
+  return (
+    <FormShell
+      resourceId={ResourceIds.PostOutwards}
+      form={formik}
+      maxAccess={access}
+      isCleared={false}
+      infoVisible={false}
+    >
+      <VertLayout>
+        <Fixed>
+          <Grid container spacing={2}>
+            <Grid item xs={12} md={6}>
+              <Grid container spacing={2}>
```

Comment:
it's necessary to have all these nested grids??

Reply:
for styling yes